sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3562,6 +3568,11 @@ def teardown_no_lock(self,
                             reason='stopping/terminating cluster nodes',
                             details=common_utils.format_exception(
                                 e, use_bracket=True)))
+                elif allow_resource_not_found:
+                    resource_not_found = _GCP_RESOURCE_NOT_FOUND_PATTERN.search(
+                        repr(e))
+                    if resource_not_found is None:
+                        raise
```

Comment:
We should handle this inside the `provisioner`. Also, can we catch the explicit exception there?

Reply:
That makes sense! If we aim to catch explicit exceptions, we would need to import the `googleapiclient` library in cloud_vm_ray_backend. That doesn't seem ideal. But incorporating this into the new provisioner shouldn't pose any issues. 🫡